app/src/main/java/com/github/orkest/View/feed/FeedActivity.kt

```diff
@@ -0,0 +1,223 @@
+ * @param song the song to display
+ */
+@Composable
+fun SongCard(song: Song){
+
+    //Row containing the song's album pic, info and play button
+        Row(horizontalArrangement = Arrangement.SpaceBetween,
+            verticalAlignment = Alignment.CenterVertically,
+            modifier = Modifier
+                .wrapContentSize()
+                .clip(shape = RoundedCornerShape(20.dp))
+                .background(Color.hsl(54f, 1f, 0.5f))
+                .padding(end = 10.dp)){
+
+            //Add the song's info at the left of the card
+            SongInfo(song)
+
+            Spacer(modifier = Modifier.width(5.dp))
+
+            //Add a play button at the right of the card
+            PlayButton()
+        }
+}
+
+@Composable
+private fun SongInfo(song: Song){
+
+    Row(Modifier.padding(10.dp)) {
+        //Add the song's picture at the left of the card
+        Image(
+            painter = painterResource(id = song.pictureId),
+            contentDescription = "Cover of the album of the song Rude Boy by Rihanna",
+            modifier = Modifier
+                .height(80.dp)
+                .clip(shape = RoundedCornerShape(10.dp))
+        )
+
+        Spacer(modifier = Modifier.width(10.dp))
+
+        Column(verticalArrangement = Arrangement.SpaceEvenly) {
+
+            //Add the song's title at the right of the album
+            Text(text = song.name, fontSize = 25.sp)
+            //Add the song's artist at the right of the album
+            Text(text = song.author, color = Color.Gray, fontSize = 18.sp)
+            //Add the song's album at the right of the album
+            Text(text = song.album, color = Color.Gray, fontSize = 14.sp)
+        }
+    }
+}
+
+@Composable
+private fun PlayButton(){
+    Icon(painter = painterResource(id = R.drawable.play_button),
+        contentDescription = "Play button",
+        modifier = Modifier
+            .height(50.dp)
+            .width(50.dp)
+            .clip(shape = RoundedCornerShape(10.dp))
+            .clickable { })
+}
+
+@Composable
+private fun Reaction(){
+    Column(modifier = Modifier.padding(20.dp)) {
+        // Create the like button
+        reactionIcon(R.drawable.black_like_icon,"Like button", "like_button" )
+        Spacer(modifier = Modifier.height(10.dp))
+
+        //Create the comment button
+        reactionIcon(R.drawable.comment_icon,"Comment button", "comment_button" )
+        Spacer(modifier = Modifier.height(10.dp))
+
+        //Create the share button
+        reactionIcon(R.drawable.share_icon,"Share button", "share_button" )
+    }
+}
+
+@Composable
+private fun reactionIcon(iconId: Int,contentDescription:String, testTag: String) {
+    Icon(painter = painterResource(id = iconId),
+        contentDescription = "Share button",
+        tint = Color.White,
+        modifier = Modifier
+            .testTag(testTag)
+            .height(20.dp)
+            .width(20.dp)
+            .clickable { })
+}
```

Comment:
I think it would be better to add an onClick parameter to this method so that you can add easily the behavior corresponding to each type of icon (Like, comment, share)  once they are implemented

Reply:
It is already done via the clickable modifier, the listener is currently empty

Reply:
But how can you adapt what's inside the clickable modifier when initiating a new reactionIcon. Wouldn't they have the same behavior since it is inside the function ?

Reply:
Ah yes I misunderstood you! Yes it will be added when the backend will be implemented